contracts/PooledStaking.sol

```diff
@@ -0,0 +1,215 @@
+      address contractAddress = staker.contracts[i];
+      uint allocation = staker.allocations[contractAddress];
+
+      uint oldAmount = oldStake.mul(allocation).div(10000);
+      uint newAmount = staker.staked.mul(allocation).div(10000);
+
+      contracts[contractAddress].staked = contracts[contractAddress].staked
+      .sub(oldAmount)
+      .add(newAmount);
+    }
+  }
+
+  function setAllocations(address[] calldata _contracts, uint[] calldata _allocations) onlyMembers external {
+
+    Staker storage staker = stakers[msg.sender];
+
+    require(
+      _contracts.length >= staker.contracts.length,
+      "Allocating to fewer contracts is not allowed"
+    );
+
+    uint previousLength = staker.contracts.length;
+    uint allocationTotal;
+
+    for (uint i = 0; i < _contracts.length; i++) {
+
+      require(_allocations[i] >= MIN_STAKE_PERCENTAGE, "Allocation minimum not met");
+      require(_allocations[i] <= 10000, "Cannot allocate more than 100% per contract");
+
+      address contractAddress = _contracts[i];
+      uint oldAllocation = staker.allocations[contractAddress];
+
+      if (i < previousLength) {
+        // we expect new contracts to be at the end
+        require(contractAddress == staker.contracts[i], "Unexpected contract");
+        require(oldAllocation <= _allocations[i], "New allocation is less than previous allocation");
+      } else {
+        staker.contracts.push(contractAddress);
+        contracts[contractAddress].stakers.push(msg.sender);
+      }
+
+      allocationTotal = allocationTotal.add(_allocations[i]);
+      uint oldAmount = staker.staked.mul(oldAllocation).div(10000);
+      uint newAmount = staker.staked.mul(_allocations[i]).div(10000);
+
+      staker.allocations[contractAddress] = _allocations[i];
+      contracts[contractAddress].staked = contracts[contractAddress]
+      .staked
+      .sub(oldAmount)
+      .add(newAmount);
+    }
+
+    require(allocationTotal <= MAX_LEVERAGE, "Total allocation exceeds maximum allowed");
+  }
+
+  function requestDeallocation(address contractAddress, uint amount) external onlyMembers {
+
+    Staker storage staker = stakers[msg.sender];
+    uint allowed = staker.allocations[contractAddress];
+
+    allowed;
+
+    require(
+      stakers[msg.sender].allocations[contractAddress] >= amount,
+      "Unable to deallocate more than allocated"
+    );
+
+    revert("NOT IMPLEMENTED");
+  }
+
+  function burn(address contractAddress, uint amount) onlyInternal external {
+    contractAddress;
+    amount;
+    revert("NOT IMPLEMENTED");
+  }
+
+  function reward(address contractAddress, uint amount) onlyInternal external {
+    contractAddress;
+    amount;
+    revert("NOT IMPLEMENTED");
+  }
+
+  function deallocate(address contractAddress, uint amount) onlyInternal public {
+    contractAddress;
+    amount;
+    revert("NOT IMPLEMENTED");
+  }
+
+  function updateParameter(ParamType param, uint value) onlyGoverned external {
```

Comment:
should this be called `updateUintParameter` or `updateUint256Parameter` in case another type is added